specs/ThrottlingControlScript.md

```diff
@@ -0,0 +1,325 @@
+  /// application, but the effective value will be constrained by resource and
+  /// platform limitations. Setting a value of `0` means a preference of no
+  /// throttling to be applied. The default value is a constant determined by
+  /// the running version of the WebView2 Runtime.
+  [propget] HRESULT ThrottlingIntervalPreferenceIntensive([out, retval] UINT32* value);
+  /// Sets the `ThrottlingIntervalPreferenceIntensive` property.
+  [propput] HRESULT ThrottlingIntervalPreferenceIntensive([in] UINT32 value);
+
+  /// The preferred wake up interval (in milliseconds) to use for throttleable
+  /// JavaScript tasks (`setTimeout` and `setInterval`), in frames whose
+  /// `ShouldUseIsolatedThrottling` property is set to `TRUE`. This is a category
+  /// specific to WebView2 with no corresponding state in the Chromium tab state
+  /// model.
+  ///
+  /// A wake up interval is the amount of time that needs to pass before the
+  /// WebView2 Runtime checks for new timer tasks to run.
+  ///
+  /// The WebView2 Runtime will try to respect the preferred interval set by the
+  /// application, but the effective value will be constrained by resource and
+  /// platform limitations. Setting a value of `0` means a preference of no
+  /// throttling to be applied. The default value is a constant determined by
+  /// the running version of the WebView2 Runtime.
+  ///
+  /// For example, an application might use an isolated throttling value of 30
+  /// ms to reduce resource consumption from third party frames in the WebView.
+  [propget] HRESULT ThrottlingIntervalPreferenceIsolated([out, retval] UINT32* value);
+  /// Sets the `ThrottlingIntervalPreferenceIsolated` property.
+  [propput] HRESULT ThrottlingIntervalPreferenceIsolated([in] UINT32 value);
+}
+
+/// A continuation of the `ICoreWebView2Frame` interface to support
+/// ShouldUseIsolatedThrottling property.
+[uuid(5b7d1b96-699b-44a2-b9f1-b8e88f9ac2be), object, pointer_default(unique)]
+interface ICoreWebView2Frame6 : ICoreWebView2Frame5 {
+  /// Indicates whether the frame has been marked for isolated throttling by the
+  /// host app. When `TRUE`, the frame will receive the throttling interval set
+  /// by `ThrottlingIntervalPreferenceIsolated`. When `FALSE`, and for main
+  /// frame, throttling interval will be determined by page state and the
+  /// interval through their corresponding properties in the
+  /// `CoreWebView2Settings` object. Defaults to `FALSE` unless set otherwise.
+  [propget] HRESULT ShouldUseIsolatedThrottling([out, retval] BOOL* value);
+
+  /// Sets the `ShouldUseIsolatedThrottling` property.
+  [propput] HRESULT ShouldUseIsolatedThrottling([in] BOOL value);
+}
+
+```
+
+```C#
+namespace Microsoft.Web.WebView2.Core
+{
+    runtimeclass CoreWebView2
+    {
+        // ...
+
+        [interface_name("Microsoft.Web.WebView2.Core.ICoreWebView2Settings9")]
+        {
+            UInt32 ThrottlingIntervalPreferenceForeground { get; set; };
```

Comment:
**For review discussion**
I'm not aware of an available COM TimeSpan-equivalent type. I can think of the following options:

* Keep current name pattern. There's likely precedent in [CoreWebView2Notification.TimeStamp](https://learn.microsoft.com/en-us/microsoft-edge/webview2/reference/win32/icorewebview2experimentalnotification?view=webview2-1.0.2194-prerelease#get_timestamp). Other `TimeStamp` APIs in Webview2 do put units in docs rather than the name: [CoreWebView2Texture.TimeStamp](https://learn.microsoft.com/en-us/microsoft-edge/webview2/reference/win32/icorewebview2experimentaltexture?view=webview2-1.0.2194-prerelease#put_timestamp) (microseconds)
* Use "Milliseconds" in the name. For discussion -- which position and whether to use "Milliseconds" or "InMilliseconds": `PreferredThrottlingInterval<1>Foreground<2>`

In either case, we can keep `UInt32` in COM and expose as `TimeSpan` in .NET/WinRT.